packages/react/src/React.ml

```diff
@@ -570,6 +572,19 @@ let useState (make_initial_value : unit -> 'state) =
   in
   (initial_value, setState)
 
+type ('input, 'output) callback = 'input -> 'output
+
+let useSyncExternalStore ~subscribe:_ ~getSnapshot:_ =
+  raise
+    (Not_implemented
+       "Do not use useSyncExternalStore on universal context, use \
+        useSyncExternalStoreWithServer instead. Read more about it \
+        https://github.com/facebook/react/blob/603e6108f39c6663ec703eed34a89ff1bf0cb70c/packages/react-server/src/ReactFizzHooks.js#L561-L566")
```

Comment:
What do you think about add a deprecation alert instead of raising an exception and return React.null?

Reply:
This way, there's no need to add Not_implemented

Reply:
@davesnx changed